tools/schematics/testing.ts

```diff
@@ -0,0 +1,136 @@
+import { ChildProcess, exec, execSync } from 'child_process';
+import { prompt } from 'enquirer';
+import fs from 'fs';
+import glob from 'glob';
+import path from 'path';
+import semver from 'semver';
+
+let currentVersion;
+
+function startVerdaccio(): ChildProcess {
+  console.log('Starting verdaccio');
+  execSync('rm -rf ./scripts/install/storage');
+  const res = exec('verdaccio --config ./scripts/install/config.yaml');
+  console.log('Pointing npm to verdaccio');
+  execSync(`npm config set @spartacus:registry http://localhost:4873/`);
+  return res;
+}
+
+function beforeExit() {
+  console.log('Setting npm back to npmjs.org');
+  execSync(`npm config set @spartacus:registry https://registry.npmjs.org/`);
+  if (verdaccioProcess) {
+    try {
+      console.log('Killing verdaccio');
+      verdaccioProcess.kill();
+    } catch {}
+  }
+}
+
+function publishLibs() {
+  if (!currentVersion) {
+    currentVersion = semver.parse(
+      JSON.parse(fs.readFileSync('projects/core/package.json', 'utf-8')).version
+    );
+  }
+
+  // Bump version to publish
+  semver.inc(currentVersion, 'patch');
+  // Packages released from it's source directory
+  const files = [
+    'projects/storefrontstyles/package.json',
+    'projects/schematics/package.json',
+  ];
+  const distFiles = glob.sync(`dist/!(node_modules)/package.json`);
+
+  [...files, ...distFiles].forEach((packagePath) => {
+    // Update version in package
+    const content = JSON.parse(fs.readFileSync(packagePath, 'utf-8'));
+    content.version = currentVersion.version;
+    fs.writeFileSync(packagePath, JSON.stringify(content, undefined, 2));
+
+    // Publish package
+    const dir = path.dirname(packagePath);
+    console.log(`\nPublishing ${content.name}`);
+    execSync(
+      `yarn publish --cwd ${dir} --new-version ${currentVersion.version} --registry=http://localhost:4873/ --no-git-tag-version`,
+      { stdio: 'inherit' }
+    );
+  });
+}
+
+function buildLibs() {
+  execSync('yarn build:libs', { stdio: 'inherit' });
+}
+
+function buildSchematics() {
+  execSync('yarn build:schematics', { stdio: 'inherit' });
+}
+
+async function executeCommand(
+  command: 'publish' | 'build schematics' | 'build all libs'
+): Promise<void> {
+  switch (command) {
+    case 'publish':
+      publishLibs();
+      break;
+    case 'build schematics':
+      buildSchematics();
+      break;
+    case 'build all libs':
+      buildLibs();
+      break;
+    default:
+      const a: never = command;
+      throw new Error(`Command ${a} not covered!`);
```

Comment:
Why not just "throw new Error(`Command ${command} not covered!`);"?

Reply:
I want also TS to scream at you and that why I needed variable with type `never` so it will fail, because you would not be possible to assign some `string` to `never`